rules/common/build.gradle.kts

```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 dependencies {
-    api(libs.kotlin.compiler)
```

Comment:
I don't think this will work well when generating the fat jars for cli / ide plugins for detekt and ktlint.

Reply:
Good point. I think that I can add separate Gradle configuration for shadowJar, but it will require adding transitive dependencies explicitly.

Example for detekt rules:

```kotlin
dependencies {
    api(libs.detekt.core)
    api(projects.rules.common)
    shadowJarInclude(libs.kotlin.compiler)
    // ...
}
```

What do you think?

Reply:
Compiling with shadowjar can be detected in this project by the existence of the "uberJar" property.

In the build.gradle.kts for the detekt / ktlint projects, we are already checking it, so I guess we could do it in the dependencies block as well:
```
if (project.hasProperty("uberJar")) { 
  api(...)
} else {
  compileOnly(...)
}
```

Reply:
I've added custom configuration that checks if `uberJar` property is set and adds dependency as `api` and `compileOnly` otherwise.

Reply:
I've fixed test dependencies, the check should pass now